zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/transform/DeploymentTransformer.java

```diff
@@ -73,6 +72,10 @@ public DeploymentTransformer(
         new FormResourceTransformer(
             keyGenerator, stateWriter, checksumGenerator, processingState.getFormState());
 
+    defaultResourceTransformer =
+        new ResourceTransformer(
+            keyGenerator, stateWriter, checksumGenerator, processingState.getResourceState());
```

Comment:
❌ The newly-implemented `ResourceTransformer` is specific to JSON files, and we should refactor it to be specific for RPA/Robot scripts. We can't make that assumption that this is the default for all deployed (non-BPMN/DMN/Form/XML) files.